helix-core/src/test/java/org/apache/helix/integration/multizk/MultiZkTestBase.java

```diff
@@ -0,0 +1,171 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import com.google.common.collect.ImmutableList;
+import org.apache.helix.HelixAdmin;
+import org.apache.helix.SystemPropertyKeys;
+import org.apache.helix.TestHelper;
+import org.apache.helix.integration.manager.ClusterControllerManager;
+import org.apache.helix.integration.manager.MockParticipantManager;
+import org.apache.helix.msdcommon.constant.MetadataStoreRoutingConstants;
+import org.apache.helix.msdcommon.mock.MockMetadataStoreDirectoryServer;
+import org.apache.helix.participant.StateMachineEngine;
+import org.apache.helix.participant.statemachine.StateModelFactory;
+import org.apache.helix.task.TaskStateModelFactory;
+import org.apache.helix.zookeeper.api.client.HelixZkClient;
+import org.apache.helix.zookeeper.api.client.RealmAwareZkClient;
+import org.apache.helix.zookeeper.datamodel.serializer.ZNRecordSerializer;
+import org.apache.helix.zookeeper.impl.factory.DedicatedZkClientFactory;
+import org.apache.helix.zookeeper.zkclient.ZkServer;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+
+import java.util.*;
+
+/**
+ * This class sets up the clusters and zk servers for multiple zk server testing.
+ */
+public class MultiZkTestBase {
+    protected static final int NUM_ZK = 3;
+    protected static final Map<String, ZkServer> ZK_SERVER_MAP = new HashMap<>();
+    protected static final Map<String, HelixZkClient> ZK_CLIENT_MAP = new HashMap<>();
+    protected static final Map<String, ClusterControllerManager> MOCK_CONTROLLERS = new HashMap<>();
+    protected final Set<MockParticipantManager> MOCK_PARTICIPANTS = new HashSet<>();
+    protected static final List<String> CLUSTER_LIST =
+            ImmutableList.of("CLUSTER_1", "CLUSTER_2", "CLUSTER_3");
+
+    protected MockMetadataStoreDirectoryServer _msds;
+    protected static final Map<String, Collection<String>> _rawRoutingData = new HashMap<>();
+    protected RealmAwareZkClient _zkClient;
+    protected HelixAdmin _zkHelixAdmin;
+
+    // Save System property configs from before this test and pass onto after the test
+    protected final Map<String, String> _configStore = new HashMap<>();
+
+    protected static final String ZK_PREFIX = "localhost:";
+    protected static final int ZK_START_PORT = 8977;
```

Comment:
nit: Just curious for the port number used. I have mostly seen port number being used in range of 21xx or 129xx range.

Reply:
Just to differentiate them with single Zk server setup which uses the ports you mentioned above